lib/ui/viewer/search/collection_suggestions.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:flutter/material.dart';
+import 'package:photos/db/files_db.dart';
+import 'package:photos/ente_theme_data.dart';
+import 'package:photos/models/collection.dart';
+import 'package:photos/models/collection_items.dart';
+import 'package:photos/models/file.dart';
+import 'package:photos/services/collections_service.dart';
+import 'package:photos/ui/viewer/file/thumbnail_widget.dart';
+import 'package:photos/ui/viewer/gallery/collection_page.dart';
+import 'package:photos/utils/navigation_util.dart';
+
+class CollectionSuggestions {
+  final List<Collection> matchedCollections;
+  final BuildContext context;
+  const CollectionSuggestions(this.matchedCollections, this.context);
+
+  List<Widget> getSuggestions() {
+    List<Widget> collectionSuggestionWidgets = [];
+    collectionSuggestionWidgets =
+        generateSuggestionWidgets(collectionSuggestionWidgets);
+    return collectionSuggestionWidgets;
+  }
+
+  List<Widget> generateSuggestionWidgets(
+    List<Widget> collectionSuggestionWidgets,
+  ) {
+    Future<List<File>> latestCollectionFiles =
+        CollectionsService.instance.getLatestCollectionFiles();
+    for (Collection collection in matchedCollections) {
+      CollectionWithThumbnail c;
+      collectionSuggestionWidgets.add(
```

Comment:
Everything inside add, including `GestureDetector` should be part of a separate widget. `CollectionSearchResultWidget` or something similar. 
We can actually remove `CollectionSuggestions` completely.  Can discuss more about this on call/in-person if required.